### PR TITLE
feat(markuplint): require command attribute on button elements

### PR DESCRIPTION
**Pull request by YusukeHirao**
## Summary

Add Markuplint rule to require `command` attribute on `button[type=button]` elements, enforcing Invoker Commands API usage for accessible UI implementation.

### 1. New Markuplint Rules

- **Require command attribute on `button[type=button]`**
  - Selector: `button[type=button]:not([role]):not([popovertarget])`
  - Exceptions: buttons with role attribute, form buttons (submit/reset/no type)
  
- **Encourage popovertarget migration**
  - Selector: `button[popovertarget]:not([command]):not([role]):not([type=submit]):not([type=reset])`
  - Displays dedicated message recommending Invoker Commands API migration

### 2. Documentation

- Updated README to document all included rules in the config
- Added comprehensive rule descriptions with examples

### 3. Tests

- Added comprehensive test cases covering:
  - type="button" buttons requiring command attribute
  - Exception cases (role attributes, form buttons)
  - popovertarget migration warnings
  - Edge cases and multiple attribute combinations

## Test Plan

- ✅ All 109 tests passing
- ✅ Button Command Attribute test validates all scenarios
- ✅ Existing tests remain unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/@d-zero/markuplint-config/base.js [hunk 1]
```diff
@@ -106,5 +106,26 @@ export default {
...
+				'required-attr': {
+					value: 'command',
+					reason:
+						'button要素には原則としてcommand属性が必要です。Invoker Commands APIを使用してアクセシブルなUIを実装してください。role属性を持つボタン（role="tab"など）やtype="submit"/type="reset"/typeなしのボタンは例外として許可されます。（D-ZERO独自ルール）',
+				},
+			},
+		},
+		{
+			selector:
+				'button[popovertarget]:not([command]):not([role]):not([type=submit]):not([type=reset])',
```
Comment (arktds) [inline packages/@d-zero/markuplint-config/base.js]: ここはシンプルに `'button[popovertarget]'` だけで良さそうです。
`command` 属性や `role` 属性などの有無に関わらず、ここでは `popovertarget` 属性の代わりに `commandfor` 属性の使用を促すだけなので。

File: packages/@d-zero/markuplint-config/base.js
Comment (arktds) [inline packages/@d-zero/markuplint-config/base.js:1]: `popovertarget` 属性の代わりに `commandfor` 属性の使用を促すルールと同様に、`popovertargetaction` 属性の代わりに `command` 属性の使用を促すルールを作成したほうが良さそうです。

File: packages/@d-zero/markuplint-config/base.js [hunk 2]
```diff
@@ -106,5 +106,26 @@ export default {
...
+			rules: {
+				'required-attr': {
+					value: 'command',
+					reason:
+						'popovertarget属性の代わりにInvoker Commands API（commandfor/command属性）の使用を推奨します。commandfor属性とshow-popover/hide-popover/toggle-popoverコマンドを使用してください。popovertargetは将来的に非推奨となる予定です。（D-ZERO独自ルール）',
```
Comment (arktds) [inline packages/@d-zero/markuplint-config/base.js]: reason も `'popovertarget属性の代わりにcommandfor属性を使用してください。popovertarget属性は将来的に非推奨となる予定です。（D-ZERO独自ルール）'` だとどうでしょうか？

File: packages/@d-zero/markuplint-config/base.js
Comment (arktds) [reply packages/@d-zero/markuplint-config/base.js:1]: reason は `'popovertargetaction属性（show/hide/toggle）の代わりにcommand属性（show-popover/hide-popover/toggle-popover）を使用してください。（D-ZERO独自ルール）'` だとどうでしょうか？

File: packages/@d-zero/markuplint-config/base.js [hunk 3]
```diff
@@ -106,5 +106,26 @@ export default {
 				},
 			},
 		},
+		{
+			selector: 'button[type=button]:not([role]):not([popovertarget])',
```
Comment (arktds) [inline packages/@d-zero/markuplint-config/base.js:110]: 後続のルールで `popovertarget` 属性の代わりに `commandfor` 属性の使用を促していますが、ここのルールでは `:not([popovertarget])` で `popovertarget` 属性が指定されている場合は許可していて良いと思います👍

File: test/cli.spec.mjs
Comment (arktds) [inline test/cli.spec.mjs:1]: 今回追加したルール以外の既存のエラーも出力されているので、それらのエラーが出ないようテストファイルを更新するとシンプルになりそうです！

**Comment by YusukeHirao** [comment]
@arktds レビュー内容、修正反映しています。

File: test/fixtures/markuplint/button-command.html [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	<!-- ========================================
+	     Should PASS: Buttons with command
+	     ======================================== -->
+
+	<!-- Basic command usage -->
+	<button commandfor="dialog" command="show-modal">Open Dialog</button>
+	<button commandfor="dialog" command="close">Close Dialog</button>
+	<button commandfor="popover" command="show-popover">Show Popover</button>
+	<button commandfor="popover" command="hide-popover">Hide Popover</button>
+	<button commandfor="details" command="toggle">Toggle Details</button>
```
Comment (arktds) [inline test/fixtures/markuplint/button-command.html]: `command="toggle"` から ``command="toggle-popover"`` に修正してください！

File: test/cli.spec.mjs [hunk 5]
```diff
@@ -183,6 +183,31 @@ describe('markuplint', () => {
...
+
+	test('Button Command Attribute', async () => {
+		const violations = await markuplint(
+			'test/fixtures/markuplint/button-command.html',
+			'packages/@d-zero/markuplint-config/base.js',
+		);
+		expect(violations).toStrictEqual([
+			'test/fixtures/markuplint/button-command.html:27:17 The "btn" class name is unmatched with the below patterns: "/^c-(?<ComponentName>[a-z][a-z0-9]*(?:-[a-z0-9]+)*)$/"',
+			'test/fixtures/markuplint/button-command.html:45:58 The "btn" class name is unmatched with the below patterns: "/^c-(?<ComponentName>[a-z][a-z0-9]*(?:-[a-z0-9]+)*)$/"',
+			'test/fixtures/markuplint/button-command.html:42:40 The "command" attribute expects either "toggle-popover", "show-popover", "hide-popover", "close", "show-modal". Or, the "command" attribute expects the custom command format. Did you mean "--toggle"? (https://html.spec.whatwg.org/multipage/form-elements.html#valid-custom-command)',
```
Comment (arktds) [inline test/cli.spec.mjs]: `command="toggle"` から `command="toggle-popover"` に修正後、このエラー文は削除してください！

**Review by arktds — CHANGES_REQUESTED** [review]

File: test/fixtures/markuplint/button-command.html (hunk 4 above)
Comment (YusukeHirao) [reply test/fixtures/markuplint/button-command.html]: 修正しました！ `command="toggle"` → `command="toggle-popover"` に変更しています。

File: test/cli.spec.mjs (hunk 5 above)
Comment (YusukeHirao) [reply test/cli.spec.mjs]: 修正しました！ `command="toggle-popover"` への修正に伴い、該当のエラー行を削除しています。

**Comment by arktds** [comment]
LGTM🎉
